File: Transact.Data/Models/Widget.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Transact.Data.Models
{
    public partial class Widget
    {
        public Widget()
        {
            WidgetAccesses = new HashSet<WidgetAccess>();
        }

        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Guid? CreatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public Guid? UpdatedBy { get; set; }

        public virtual ICollection<WidgetAccess> WidgetAccesses { get; set; }
    }
}

File: Transact.Web/src/app/administration/widgets/widget.ts
import { Role } from '../useraccounts/useraccount';

export class Widget {
  Id: string | null = null;
  public Name: string | null = null;
  public CreatedOn: Date | null = null;
  public CreatedBy: number | null = null;
  public UpdatedOn: Date | null = null;
  public UpdatedBy: number | null = null;
  public Roles: Role[] = [];
}
